### PR TITLE
chore: add catalog info and renovate.json

### PR DESCRIPTION
**Pull request by ichintanjoshi**
## Description

- Add catalog-info.yaml
- Add renovate.json

### CODE REVIEW

**Comment by openedx-webhooks** [comment]
Thanks for the pull request, @ichintanjoshi! Please note that it may take us up to several weeks or months to complete a review and merge your PR.

Feel free to add as much of the following information to the ticket as you can:
- supporting documentation
- [Open edX discussion forum threads](https://discuss.openedx.org/)
- timeline information ("this must be merged by XX date", and why that is)
- partner information ("this is a course on edx.org")
- any other information that can help Product understand the context for the PR

All technical communication about the code itself will be done via the GitHub pull request interface. As a reminder, [our process documentation is here](http://edx-developer-guide.readthedocs.org/en/latest/process/overview.html).


Please let us know once your PR is ready for our review and all tests are green.

File: .github/renovate.json
Comment (feanil) [inline .github/renovate.json:1]: Since this repo doesn't have a `package.json` this is not needed.

File: catalog-info.yaml [hunk 1]
```diff
@@ -0,0 +1,21 @@
...
+  description: >
+            This provides a student with hints when they
+            incorrectly answer a problem within a course
+  links:
+    - url: "https://github.com/openedx/crowdsourcehinter"
+      title: "openedx/crowdsourcehinter"
+      icon: "GitHub"
+  annotations:
+    openedx.org/arch-interest-groups: ""
+    openedx.org/release: "main"
```
Comment (feanil) [inline catalog-info.yaml:17]:
```suggestion
```
We don't tag this directly as a part of named releases so we should not add this line.

File: catalog-info.yaml [hunk 2]
```diff
@@ -0,0 +1,21 @@
...
+kind: Component
+metadata:
+  name: 'crowdsourcehinter'
+  description: >
+            This provides a student with hints when they
+            incorrectly answer a problem within a course
+  links:
+    - url: "https://github.com/openedx/crowdsourcehinter"
+      title: "openedx/crowdsourcehinter"
+      icon: "GitHub"
```
Comment (feanil) [inline catalog-info.yaml:14]: This link is not necessary, the repo is automatically linked in Backstage already.

File: .github/renovate.json
Comment (ichintanjoshi) [reply .github/renovate.json:1]: Have removed it

File: catalog-info.yaml (hunk 2 above)
Comment (ichintanjoshi) [reply catalog-info.yaml]: updated

**Review by feanil — APPROVED** [review]

**Comment by openedx-webhooks** [comment]
@ichintanjoshi 🎉 Your pull request was merged!  Please take a moment  to [answer a two question survey](https://docs.google.com/forms/d/e/1FAIpQLSceJOyGJ6JOzfy6lyR3T7EW_71OWUnNQXp68Fymsk3MkNoSDg/viewform?usp=pp_url&entry.1671973413=openedx/crowdsourcehinter&entry.867055334=https://github.com/openedx/crowdsourcehinter/pull/155&entry.1484655318=https://github.com/ichintanjoshi&entry.752974735=2024-05-30+07:32&entry.1917517419=2024-06-20+17:42&entry.2133058324=Yes) so we can improve your experience in the future.

**Pull request merged**